src/api/deploy/helpers/lookup-tenant-service.js

```diff
@@ -1,42 +1,63 @@
 import { getContent } from '~/src/helpers/github/get-content'
 import { config } from '~/src/config'
+import { createLogger } from '~/src/helpers/logging/logger'
+import Joi from 'joi'
 
-async function lookupTenantService(service, environment, logger) {
-  const filePath = `environments/${environment}/tenants/${service}.json`
-  const owner = config.get('github.org')
-  const repo = config.get('github.repos.cdpTfSvcInfra')
+const logger = createLogger()
+
+const org = config.get('github.org')
+const repo = config.get('github.repos.cdpTfSvcInfra')
+
+const schema = Joi.object({
+  zone: Joi.string().valid('public', 'protected').required(),
+  service_code: Joi.string().min(3).required(),
+  mongo: Joi.boolean(),
+  redis: Joi.boolean(),
+  testSuite: Joi.string()
+}).unknown(true)
 
+/**
+ * Get service from the multi-file version of tenants.json
+ * @param {string} service
+ * @param {string} environment
+ * @param {string} ref
+ * @returns {Promise<undefined|*>}
+ */
+async function lookupTenantService(service, environment, ref = 'main') {
+  const filePath = `environments/${environment}/tenants/${service}.json`
   try {
-    const data = await getContent(owner, repo, filePath)
+    const data = await getContent(org, repo, filePath, ref)
     const service = JSON.parse(data)
-
-    if (service?.zone && service?.service_code) {
-      return service
-    } else {
-      logger.warn(
-        `${service}.json did not contain zone and service_code - Falling back to tenant_services.json`
-      )
-      return await lookupTenantServices(
-        service,
-        environment,
-        owner,
-        repo,
-        logger
-      )
-    }
+    Joi.assert(service, schema) // Check file in correct format
```

Comment:
tbh im not 100% sold on using Joi here. kinda made sense at the time but feels like an overkill for replacing an if statement?

Reply:
Looking through the code we are expecting this to be in a specific format. I can see references to:
- `service.zone`
- `service.name`
- `service.tenantConfig.testSuite`

IIRC we did this because it needed to be in a specific format when it was being committed back to the TF after it had been updated.
The `unknown` was to stop additions in TF breaking Self Service Ops


Considering we are not committing this back to the TF now I think we can drop this. But if doing that I would make sure all references to it are guarded correctly and have debug logs, so we can see if things go wrong.

I suspect the TF will change in the future and if it does we need this work to behave well

Reply:
Yeah, we might be able to skip reading the tenant files all together and have the build pipeline just send us the tenant list after every run. It currently sends it to secrets lambda so no reason portal can't get a copy